Formula/grass.rb

```diff
@@ -0,0 +1,352 @@
+      osascript -  \"$cmd\"  <<EOF
+      on run argv -- argv is a list of strings
+      tell application \"Terminal\"
+      if not (exists window 1) then reopen
+      activate
+      do script (item 1 of argv) in window 1
+      end tell
+      end run
+    EOS
+
+    File.write("#{grass_macos_app}/Contents/MacOS/grass.sh", file)
+    chmod("+x", "#{grass_macos_app}/Contents/MacOS/grass.sh")
+
+    config = <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+      <key>CFBundleDevelopmentRegion</key>
+      <string>en</string>
+      <key>CFBundleExecutable</key>
+      <string>grass.sh</string>
+      <key>NSHumanReadableCopyright</key>
+      <string>Copyright © 1999–2022 GRASS Development Team</string>
+      <key>CFBundleIconFile</key>
+      <string>AppIcon</string>
+      <key>CFBundleName</key>
+      <string>GRASS-#{version}</string>
+      <key>CFBundleDisplayName</key>
+      <string>GRASS-#{version}</string>
+      <key>CFBundleIdentifier</key>
+      <string>org.osgeo.grass</string>
+      <key>CFBundleInfoDictionaryVersion</key>
+      <string>6.0</string>
+      <key>CFBundlePackageType</key>
+      <string>APPL</string>
+      <key>CFBundleShortVersionString</key>
+      <string>#{version}</string>
+      <key>CFBundleVersion</key>
+      <string>#{version}</string>
+      <key>CFBundleDocumentTypes</key>
+      <array>
+        <dict>
+          <key>CFBundleTypeName</key>
+          <string>GRASS Workspace File</string>
+          <key>CFBundleTypeExtensions</key>
+          <array>
+            <string>gxw</string>
+          </array>
+          <key>CFBundleTypeIconFile</key>
+          <string>GRASSDocument_gxw</string>
+        </dict>
+      </array>
+      <key>LSMinimumSystemVersion</key>
+      <string>12.3</string>
+      <key>LSHasLocalizedDisplayName</key>
+      <true/>
+        </dict>
+      </plist>
+    EOS
+
+    # File.open("#{grass_macos_app}/Contents/Info.plist", "w") { |file| file.write(config) }
+    File.write("#{grass_macos_app}/Contents/Info.plist", config)
+  end
+
+  def caveats
+    s = <<~EOS
+      If it is the case that you can change the shebang at the beginning of
+      the script to enforce Python 3 usage.
+
+        \e[32m#!/usr/bin/env python\e[0m
+
+      Should be changed into
+
+        \e[32m#!/usr/bin/env python3\e[0m
+
+    EOS
+
+    if head_only?
+      s += <<~EOS
+        This build of GRASS has been compiled without the WxPython GUI.
+
+        The command line tools remain fully functional.
+      EOS
+    end
+
+    s += <<~EOS
+      You may also copy \e[32mGRASS.app\e[0m to \e[32m/Applications\e[0m or \e[32m~/Applications\e[0m:
+
+        \e[32mcp -r `find $(brew --prefix) -name "GRASS*.app"` /Applications/GRASS.app\e[0m
+    EOS
+    s
+  end
+
+  test do
+    system bin/"grass#{majmin_ver}", "--version"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula